lib/RefBackend/RefBackend.cpp

```diff
@@ -219,77 +222,89 @@ mlir::torch::RefBackend::createMungeCallingConventionsPass() {
+  RankedTensorType tensorType = globalOp.getType().cast<RankedTensorType>();
+  MemRefType memrefType =
+      MemRefType::get(tensorType.getShape(), tensorType.getElementType());
+  b.setInsertionPointToStart(globalOp->getParentOfType<ModuleOp>().getBody());
+  if (!globalOp.getValue().has_value())
+    return emitError(globalOp.getLoc()).append("global op must have a value");
   b.create<memref::GlobalOp>(
-      UnknownLoc::get(b.getContext()), getSeedGobalVarName(),
-      /*sym_visibility=*/b.getStringAttr("private"),
-      /*type=*/memref0D,
-      /*initial_value=*/DenseIntElementsAttr::get(tensor0D, {APInt(64, 0)}),
-      /*constant=*/false,
+      UnknownLoc::get(b.getContext()), globalOp.getSymName(),
+      /*sym_visibility=*/globalOp.getSymVisibilityAttr(),
+      /*type=*/memrefType,
+      /*initial_value=*/globalOp.getValue().value(),
+      /*constant=*/globalOp.getIsMutable() ? false : true,
       /*alignment=*/nullptr);
+  return success();
 }
 
-// Generate sequence for getting the next seed with LCG step:
-//    nextSeed = (multiplier * currentSeed + incrementStep) mod 64.
-// Refer to https://en.wikipedia.org/wiki/Linear_congruential_generator.
-static Value lowerGetNextSeed(OpBuilder &b, Location loc) {
-  // Get the current seed value.
-  auto memref1DType = MemRefType::get({}, b.getI64Type());
-  Value globalVar =
-      b.create<memref::GetGlobalOp>(loc, memref1DType, getSeedGobalVarName());
-  Value currentSeed = b.create<memref::LoadOp>(loc, globalVar);
-
-  // The value of multiplier and incrementStep are referenced from
-  // https://en.wikipedia.org/wiki/Linear_congruential_generator for 2^64.
-  Value multiplier = b.create<arith::ConstantOp>(
-      loc, b.getI64IntegerAttr(6364136223846793005));
-  Value incrementStep = b.create<arith::ConstantOp>(
-      loc, b.getI64IntegerAttr(1442695040888963407));
-  // temp = multiplier * currentSeed + incrementStep
-  Value mul = b.create<arith::MulIOp>(loc, currentSeed, multiplier);
-  Value temp = b.create<arith::AddIOp>(loc, mul, incrementStep);
-  // temp mod 64 = temp & 63
-  Value cst127 = b.create<arith::ConstantOp>(loc, b.getI64IntegerAttr(127));
-  Value nextSeed = b.create<arith::AndIOp>(loc, temp, cst127);
-  b.create<memref::StoreOp>(loc, nextSeed, globalVar);
-  return nextSeed;
+static LogicalResult bufferizeMLProgramGlobaLoadOp(ml_program::GlobalLoadOp globalLoadOp,
+                                          OpBuilder &b,
+                                          SmallVector<Operation *> &toErase) {
+  RankedTensorType tensorType = globalLoadOp.getType().cast<RankedTensorType>();
+  MemRefType memrefType =
+      MemRefType::get(tensorType.getShape(), tensorType.getElementType());
+  b.setInsertionPoint(globalLoadOp);
+  Value memref = b.create<memref::GetGlobalOp>(
+      globalLoadOp.getLoc(), memrefType,
+      globalLoadOp.getGlobalAttr().getLeafReference());
+  for (auto user : globalLoadOp->getResult(0).getUsers()) {
```

Comment:
this is not correct. You should separately bufferize the ml_program.global_store op to memref.copy into the global.